dts/arm/nuvoton/npcx/npcx7/npcx7-pinctrl.dtsi

```diff
@@ -0,0 +1,1001 @@
+		io-pad = <0x0c 1 0x02 4>; /* Low-voltage input supported */
+	};
+
+	ioc2_i2c6_scl0_pwm1: ioc2-cfg {
+		io-pad = <0x0c 2 0x04 1>; /* Low-voltage input supported */
+	};
+
+	ioc3_pwm0: ioc3-cfg {
+		io-pad = <0x0c 3>;
+	};
+
+	ioc4_pwm2: ioc4-cfg {
+		io-pad = <0x0c 4>;
+	};
+
+	ioc5: ioc5-cfg {
+		io-pad = <0x0c 5 0x03 7>; /* Low-voltage input supported */
+	};
+
+	ioc6_smi: ioc6-cfg {
+		io-pad = <0x0c 6 0x03 0>; /* Low-voltage input supported */
+	};
+
+	ioc7_dtr_bout: ioc7-cfg {
+		io-pad = <0x0c 7 0x02 5>; /* Low-voltage input supported */
+	};
+
+	/* IO port D */
+	iod0_i2c3_sda0: iod0-cfg {
+		io-pad = <0x0d 0 0x01 3>; /* Low-voltage input supported */
+	};
+
+	iod1_i2c3_scl0: iod1-cfg {
+		io-pad = <0x0d 1 0x01 2>; /* Low-voltage input supported */
+	};
+
+	iod2_psl_in1: iod2-cfg {
+		io-pad = <0x0d 2>;
+	};
+
+	iod3_tb1: iod3-cfg {
+		io-pad = <0x0d 3>;
+	};
+
+	iod4: iod4-cfg {
+		io-pad = <0x0d 4>;
+	};
+
+	iod5: iod5-cfg {
+		io-pad = <0x0d 5>;
+	};
+
+	iod6: iod6-cfg {
+		io-pad = <0x0d 6>;
+	};
+
+	iod7: iod7-cfg {
+		io-pad = <0x0d 7>;
+	};
+
+	/* IO port E */
+	ioe0: ioe0-cfg {
+		io-pad = <0x0e 0>;
+	};
+
+	ioe1_adc7: ioe1-cfg {
+		io-pad = <0x0e 1>;
+	};
+
+	ioe2: ioe2-cfg {
+		io-pad = <0x0e 2>;
+	};
+
+	ioe3_i2c6_sda1: ioe3-cfg {
+		io-pad = <0x0e 3 0x04 7>; /* Low-voltage input supported */
+	};
+
+	ioe4_i2c6_scl1: ioe4-cfg {
+		io-pad = <0x0e 4 0x04 6>; /* Low-voltage input supported */
+	};
+
+	ioe5: ioe5-cfg {
+		io-pad = <0x0e 5>;
+	};
+
+	/* IO port F */
+	iof0_adc9: iof0-cfg {
+		io-pad = <0x0f 0>;
+	};
+
+	iof1_adc8: iof1-cfg {
+		io-pad = <0x0f 1>;
+	};
+
+	iof2_i2c4_sda1: iof2-cfg {
+		io-pad = <0x0f 2 0x04 3>; /* Low-voltage input supported */
+	};
+
+	iof3_i2c4_scl1: iof3-cfg {
+		io-pad = <0x0f 3 0x04 2>; /* Low-voltage input supported */
+	};
+
+	iof4_i2c5_sda1: iof4-cfg {
+		io-pad = <0x0f 4 0x04 5>; /* Low-voltage input supported */
+	};
+
+	iof5_i2c5_scl1: iof5-cfg {
+		io-pad = <0x0f 5 0x04 4>; /* Low-voltage input supported */
+	};
+
+	/* Nodes for peripheral device's pin-muxing and pad properties */
+	/* SCFG DEVALT 0 */
+	periph_alt0_spip_sl: alt00-cfg {
+		pinmux = <0x00 0x0 0>;
+		periph-pupd = <0x01 6>;
+	};
+
+	periph_alt0_gpio_no_spip: alt03-inv-cfg {
+		pinmux = <0x00 0x3 1>;
+		pinmux-gpio;
+	};
+
+	periph_alt0_gpio_no_fpip: alt07-inv-cfg {
+		pinmux = <0x00 0x7 1>;
+		pinmux-gpio;
+	};
+
+	/* SCFG DEVALT 1 */
+	periph_alt1_kbrst_sl: alt10-cfg {
+		pinmux = <0x01 0x0 0>;
+	};
+
+	periph_alt1_a20m_sl: alt11-cfg {
+		pinmux = <0x01 0x1 0>;
+	};
+
+	periph_alt1_smi_sl: alt12-cfg {
+		pinmux = <0x01 0x2 0>;
+	};
+
+	periph_alt1_ec_sci_sl: alt13-cfg {
+		pinmux = <0x01 0x3 0>;
+	};
+
+	periph_alt1_no_pwrgd: alt14-inv-cfg {
+		pinmux = <0x01 0x4 1>;
+		pinmux-gpio;
+	};
+
+	periph_alt1_pwroff: alt15-cfg {
+		pinmux = <0x01 0x5 0>;
+	};
+
+	periph_alt1_clkrn_sl: alt16-cfg {
+		pinmux = <0x01 0x6 0>;
+	};
+
+	periph_alt1_no_lpc_espi: alt17-inv-cfg {
+		pinmux = <0x01 0x7 1>;
+		pinmux-gpio;
```

Comment:
Indeed, the ```pinmux-gpio``` looks ambiguous. I also got comments that the node name such as  ```periph_alt1_no_lpc_espi``` is confused in our internal code review. Hence, I'd like to arrange the peripheral pinctrl nodes as below:

```
/** User can put pinmux cells into pinmux prop. of this pinctrl node in 
 *  board DT layout file. Then, the mux of these pads will switch to GPIO 
 *  before driver initialization. It might be better to maintain.
 */
periph_gpio_def: alt-gpio-cfg {
	pinmux = <&alt1_no_lpc_espi
			&alt1_no_pwrgd
			&alt7_no_ksi0_sl>;
	pinmux-gpio;
};

periph_lpc_espi_def: periph_lpc_espi_def {
	pinmux = <&alt1_no_lpc_espi>;
};

periph_uart1_1_def: periph_uart1_1_def {
	pinmux = <&altj_cr_sin1_sl1
			 &altj_cr_sout1_sl1>;
};


&espi0 {
	pinctrl-0 = <&periph_lpc_espi_def>;
	pinctrl-names = "default";
};

&uart1 {
	pinctrl-0 = <&periph_uart1_1_def>;
	pinctrl-names = "default";
};
```

If you're okay, I will start to prepare it.

Reply:
Yes, this looks better.  `periph_gpio_def` replaces the `def-io-conf-list` in the current implementation, correct?

Reply:
Yes. Although the user can set ```pinmux-gpio``` properties in each peripheral pinctrl node one by one, it is easy to maintain to put them together. And overwrite it in the board DT layout if needed. I will prepare the PR based on the previous comment.

Reply:
Done in the new PR.